lib/gc/Dialect/OneDNNGraph/OneDNNGraphOps.cpp

```diff
@@ -65,11 +27,13 @@ LogicalResult onednn_graph::AddOp::inferReturnTypeComponents(
   auto getShapeIdx = [](ValueShapeRange operands, size_t i) {
     return operands.getShape(i);
   };
-  auto ret =
-      inferBroadcastShape<ValueShapeRange>(operands, outShape, getShapeIdx);
+  llvm::SmallVector<int64_t> input1, input2;
+  getShapeIdx(operands, 0).getDims(input1);
+  getShapeIdx(operands, 1).getDims(input2);
+  auto ret = OpTrait::util::getBroadcastedShape(input1, input2, outShape);
   inferredReturnShapes.push_back(
       ShapedTypeComponents(outShape, resultTy.getElementType()));
-  return ret;
+  return LogicalResult::success(ret);
```

Comment:
What is the difference between `return ret` and `return LogicalResult::success(ret)`

Reply:
The return value ret returned by the new function is `bool` . Here we need to convert it to `logicalResult`.

Reply:
Oh, I see, to convert bool to LogicalResult